### PR TITLE
Also compare uploaded package against packages that its dar embeds

### PR DESCRIPTION
**Pull request by paulbrauner-da**
Addresses https://github.com/digital-asset/daml/issues/19742 for 2.9.

### CODE REVIEW

**Review by samuel-williams-da — APPROVED** [review]
Looks good, thanks! Be sure to update your base branch. I'll be merging https://github.com/digital-asset/daml/pull/19750 (if its CI ever starts?), so if that gets in first, you may need to rebase again

File: sdk/canton/community/participant/src/main/scala/com/digitalasset/canton/participant/admin/PackageUpgradeValidator.scala [hunk 1]
```diff
@@ -185,8 +202,13 @@ class PackageUpgradeValidator(
       }
       .filter { case (_, version) => pkgVersion > version }
       .maxByOption { case (_, version) => version }
-      .traverse { case (pId, _) => lookupDar(pId) }
-      .map(_.flatten)
+      .traverse { case (pId, _) =>
+        lookupDar(pId, upgradingPackagesMap)(loggingContext).map(
+          _.getOrElse(
+            throw new IllegalStateException("failed to look up dar present in package map")
```
Comment (remyhaemmerle-da) [inline sdk/canton/community/participant/src/main/scala/com/digitalasset/canton/participant/admin/PackageUpgradeValidator.scala:208]: In general we should log internal errors. 
A cheap way is to extend `com.digitalasset.daml.lf.InternalError`.

**Review by remyhaemmerle-da — APPROVED** [review]
thanks for the quick fix

File: sdk/canton/community/participant/src/main/scala/com/digitalasset/canton/participant/admin/PackageUpgradeValidator.scala (hunk 1 above)
Comment (paulbrauner-da) [reply sdk/canton/community/participant/src/main/scala/com/digitalasset/canton/participant/admin/PackageUpgradeValidator.scala]: Done

**Pull request merged**